internal/util/update/resource-merge_test.go

```diff
@@ -0,0 +1,134 @@
+package update
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"sigs.k8s.io/kustomize/kyaml/copyutil"
+	"sigs.k8s.io/kustomize/kyaml/openapi"
+)
+
+func TestMergeSubPackages(t *testing.T) {
+	var tests = []struct {
+		name           string
+		updatedDataSet string
+		localDataset   string
+	}{
+		{
+			name:           "autoset-recurse-subpackages",
```

Comment:
This seems like a pretty complex flow, so isn't there other scenarios we should also test?

Reply:
I tried to comment as much as possible to make the flow clear. I tried to cover all the scenarios in a flow.